src/sage/rings/polynomial/polynomial_singular_interface.py

```diff
@@ -333,8 +336,10 @@ def _singular_(self, singular=singular):
         """
         try:
             R = self.__singular
-            if not (R.parent() is singular):
+            if singular is not None and R.parent() is not singular:
                 raise ValueError
+            elif singular is None:
+                from sage.interfaces.singular import singular
             R._check_valid()
```

Comment:
Perhaps line 339 and line 341 should be reversed. Also note that in line 45, `singular` is already imported...

Reply:
Done in f90914eaad9

> note that in line 45, `singular` is already imported...

Removed now in ece9c930230